### PR TITLE
Refactor code from sample data files to repositories

### PR DESCRIPTION
**Pull request by shubhamsinghshubham777**
This PR deals with the refactoring of the sample data methods `like getSampleServer(), getSampleServerList(), etc` to their respective repositories.

Previously, these methods were being called directly from the composables making it hard to refactor when writing their real implementations, but now since we have all the implementations inside the respective repository implementations inside an extension function called `safeApiCall` (which has 2 params i.e. `debugResponse` and `releaseResponse`), therefore we now only have to write the real implementations inside the `releaseResponse` block and the app should behave just like before (but now with real data).

`Note: These methods are still being called from some of the composable PREVIEWS directly since composable previews don't support rendering with ViewModels yet.`

### CODE REVIEW

File: app/src/main/java/dev/baseio/discordjetpackcompose/ui/routes/dashboard/DashboardRoute.kt [hunk 1]
```diff
@@ -23,7 +26,20 @@ fun NavGraphBuilder.dashboardRoute(
         route = DiscordRoute.Dashboard.name
     ) {
         composable(DiscordScreen.Dashboard.name) {
-            DashboardScreen(composeNavigator = composeNavigator)
+            val dashboardScreenVM = hiltViewModel<DashboardScreenViewModel>()
+
+            LaunchedEffect(Unit) {
```
Comment (Anmol92verma) [inline app/src/main/java/dev/baseio/discordjetpackcompose/ui/routes/dashboard/DashboardRoute.kt:31]: @shubhamsinghshubham777 can we move this around

File: app/src/main/java/dev/baseio/discordjetpackcompose/viewmodels/DashboardScreenViewModel.kt [hunk 2]
```diff
@@ -1,27 +1,99 @@
...
+                val result = getServerUseCase(serverId = serverId)
+                if (result is NetworkState.Success) {
+                    searchSheetServerList.add(result.data)
+                }
+            }
+        }
+    }
+
+    fun getSearchSheetItemList() {
+        ioScope {
```
Comment (Anmol92verma) [inline app/src/main/java/dev/baseio/discordjetpackcompose/viewmodels/DashboardScreenViewModel.kt:62]: @shubhamsinghshubham777 we can get rid of iOscope https://developer.android.com/kotlin/coroutines/coroutines-best-practices

**Comment by Anmol92verma** [comment]
@shubhamsinghshubham777 lmk once ready to be reviewed again for merging ?

**Comment by shubhamsinghshubham777** [comment]
@Anmol92verma PR is ready to review now

**Pull request merged**